docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,540 @@
+It could be done using initial "from" array by checking every combination against "from" items.
+
+Example
+
+We have a "from" array:
+
+```yaml
+from:
+  - targetRef:
+      kind: Mesh
+    default:
+      action: ALLOW
+  - targetRef: 
+      kind: MeshSubset
+      tags:
+        zone: us-east
+    default:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        env: dev
+    default:
+      action: ALLOW
+  - targetRef:
+       kind: MeshSubset
+       tags:
+          env: prod
+    default:
+       action: ALLOW
+```
+
+1. Create "allTags" array:
+```go
+allTags = []Tag{
+	{key: "zone", value: "us-east"},
+	{key: "env", value: "dev"},
+	{key: "env", value: "prod"},
+	
+}
+```
+
+2. Create all possible combination with negations:
+
+```go
+comb1 = []Tag{
+    {key: "zone", value: "us-east"},
+    {key: "env", value: "dev"},
+    {key: "env", value: "!prod"},
+}
+comb2 = []Tag{
+    {key: "zone", value: "us-east"},
+    {key: "env", value: "!dev"},
+    {key: "env", value: "prod"},
+}
+comb3 = []Tag{
+    {key: "zone", value: "!us-east"},
+    {key: "env", value: "dev"},
+    {key: "env", value: "!prod"},
+}
+comb4 = []Tag{
+    {key: "zone", value: "!us-east"},
+    {key: "env", value: "!dev"},
+    {key: "env", value: "prod"},
+}
+comb5 = []Tag{
+    {key: "zone", value: "us-east"},
+    {key: "env", value: "!dev"},
+    {key: "env", value: "!prod"},
+}
+comb6 = []Tag{
+    {key: "zone", value: "!us-east"},
+    {key: "env", value: "!dev"},
+    {key: "env", value: "!prod"},
+}
+comb7 = []Tag{ // no clients can have env=dev and env=prod at the same time 
+    {key: "zone", value: "us-east"},
+    {key: "env", value: "dev"},
+    {key: "env", value: "prod"},
+}
+comb8 = []Tag{ // no clients can have env=dev and env=prod at the same time
+    {key: "zone", value: "!us-east"},
+    {key: "env", value: "dev"},
+    {key: "env", value: "prod"},
+}
+```
+
+![img.png](assets/rules-example.png)
+
+3. Determine configuration for each combination (only number 5-green is DENY):
+
+```yaml
+rules:
+  - targetRef:
+      kind: MeshSubset
+      tags:
+         zone: us-east
+         env: dev
+    default:
+      action: ALLOW
+  - targetRef:
+       kind: MeshSubset
+       tags:
+          zone: us-east
+          env: prod
+    default:
+       action: ALLOW
+  - targetRef:
+       kind: MeshSubset
+       tags:
+          zone: !us-east
+          env: dev
+    default:
+       action: ALLOW
+  - targetRef:
+       kind: MeshSubset
+       tags:
+          zone: !us-east
+          env: prod
+    default:
+       action: ALLOW
+  - targetRef:
+       kind: MeshSubset
+       tags:
+          zone: us-east
+          env: !prod
+          env: !dev
+    default:
+       action: DENY
+  - targetRef:
+       kind: MeshSubset
+       tags:
+          zone: !us-east
+          env: !prod
+          env: !dev
+    default:
+       action: ALLOW
+```
```

Comment:
I think it'd be useful to add a step that drop all the `DENY` entries as they are not needed anymore.

Reply:
Done